poc/experiment.py

```diff
@@ -0,0 +1,214 @@
+    }
+
+    ex.add_named_config(config_name, config)
+
+
+for (dataset, config_name, dataset_type) in [
+    (
+        TrivialDiscreteChoiceProblem,
+        "trivial_discrete_choice_problem",
+        "discrete_choice",
+    ),
+    (
+        TrivialRankingProblem,
+        "trivial_ranking_problem",
+        "ranking",
+    ),
+    (
+        TrivialVariableChoiceProblem,
+        "trivial_variable_choice_problem",
+        "variable_choice",
+    ),
+    (ParetoChoiceProblem, "pareto_choice_problem", "variable_choice"),
+]:
+    config = {
+        "dataset": dataset,
+        "dataset_params": {
+            "n_features": 2,
+            "n_objects": 3,
+            "n_instances": int(1e5),
+        },
+        "dataset_type": dataset_type,
+    }
+    if dataset_type == "discrete_choice":
+        config["dataset_params"]["choice_size"] = 2
+
+    for (parameter, value) in inspect.signature(dataset).parameters.items():
+        if value.default != inspect._empty:
+            config["dataset_params"][parameter] = value.default
+    ex.add_named_config(config_name, config)
+
+
+@ex.named_config
+def learning_choice_functions_feta_pareto():
+    """A (partial) reproduction of the experimental setup of the FETA paper."""
+    # Refer to Table 2 in https://arxiv.org/abs/1901.10860 for hyperparameters
+    # and reasonable ranges.
+    estimator = FETAVariableChoiceEstimator
+    estimator_params = {
+        "max_epochs": int(1e9),  # TODO look up when to terminate in the paper
+        # TODO make this more ergonomic to configure
+        # See https://github.com/skorch-dev/skorch/pull/273#issuecomment-740908228
+        "module__pairwise_utility_module": functools.partial(
+            PairwiseDenseUtility,
+            hidden_layers=10,  # "Layers" in the paper
+            units_per_hidden=30,  # "Units" in the paper
+        ),
+        "optimizer": optim.SGD,
+        "optimizer__lr": 1e-3,  # "Learning Rate" or "lr_0" in the paper
+        "optimizer__weight_decay": 1e-6,  # "Regularizer Strength" in the paper
+        "optimizer__nesterov": True,  # fixed
+        "optimizer__momentum": 0.9,  # fixed
+        "batch_size": 512,  # "Batch Size" in the paper
+        # "scoring_module__activation": nn.SELU,  # "activation function for each hidden layer" in the paper (ReLU + batch norm / SeLU) (TODO)
+        "callbacks": [
+            # StepLR, step_size, gamma
+            # https://pytorch.org/docs/stable/optim.html
+            LRScheduler(
+                policy="StepLR",
+                step_size=175,  # "e_drop" in the paper
+                gamma=0.05,  # "d_r" in the paper
+            )
+        ],
+    }
+    estimator_type = "variable_choice"
+    dataset = ParetoChoiceProblem
+    dataset_params = {
+        "n_features": 2,
+        "n_objects": 30,
+        "n_instances": int(1e5),
+    }
+    dataset_type = "variable_choice"
+    # TODO set cross-validation parameters as in the paper
```

Comment:
I'm also not sure if I understand the training and validation strategy of ["Leaning Choice  Functions"](https://arxiv.org/pdf/1901.10860.pdf) correctly. I understand that it uses and outer 5-fold cross validation for hyper-parameter optimization, but I'm not sure how the "inner" validation works. From Table 3 I would guess that it works like this:

- Generate (110000 / 4) * 5 = 137,500 instances (to get that 100,000 + 10,000 split in each cross-validation step).
- Split this into 5 folds of 27,500 instances each.
- For each fold F (outer cross-validation loop):
	- Combine the other 4 folds into one dataset of 110,000 instances.
	- Split this into training (100,000) and test (10,000) data
	- For each set of hyperparameters
		- Train a model, test it on the test data.
	- Validate the best (according to test data) model on fold F
- Report the average performance of the outer cross-validation.

However I'm almost sure that is incorrect :smile: Could you clarify @kiudee?